src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialog.cs

```diff
@@ -345,8 +346,34 @@ private void RegisterEvents()
 				window.SizeChanged -= WindowSizeChanged;
 			});
 
+			if (SystemNavigationManager.GetForCurrentView() is { } navManager)
```

Comment:
I wonder if UWP did it that way... We would have to test on Win10 which still has that mode :)

Reply:
They don't do it this way, ideally the BackRequested event is never fired in this scenario. We would need to have some sort of new internal event that gets fired on back pressed, and then BackRequested would only fire if that new event's Handled was still false

Reply:
Makes sense. Let's explain this in the code as a comment.